implemented AuthenticationService;

knyazkyylubomir opened the pull request:
implemented AuthenticationController;
implemented CustomUserDetailsService;
added FieldMatch annotation;
implemented FieldMatchValidator;
implemented JwtAuthenticationFilter;
implemented JwtUtil;
implemented UserServiceImpl;
added DuplicateException and RegistrationException handlers to CustomGlobalExceptionHandler;
added checkIfIsbnIsDuplicate to BookServiceImpl;
added PreAuthorized annotation over particular endpoints in BookController;

knyazkyylubomir committed:
implemented BookController;
implemented BookMapper;
added findBookById() to BookRepositoryImpl;
added getBookById() to BookServiceImpl;
implemented MapperConfig

knyazkyylubomir committed:
replaced BookRepositoryImpl with JpaRepository API;
added updateBookById(), searchBook() and deleteById() in BookServiceImpl;
added required endpoints in BookController;
implemented BookSpecificationBuilder;
added liquibase support

knyazkyylubomir committed:
added minor fix

knyazkyylubomir committed:
implemented CustomGlobalExceptionHandler;
added validation annotations over fields in CreateBookRequestDto

knyazkyylubomir committed:
added pagination, sorting and swagger-ui

knyazkyylubomir committed:
implemented AuthenticationService;
implemented AuthenticationController;
implemented CustomUserDetailsService;
added FieldMatch annotation;
implemented FieldMatchValidator;
implemented JwtAuthenticationFilter;
implemented JwtUtil;
implemented UserServiceImpl;
added DuplicateException and RegistrationException handlers to CustomGlobalExceptionHandler;
added checkIfIsbnIsDuplicate to BookServiceImpl;
added PreAuthorized annotation over particular endpoints in BookController;

kozhukhovsky commented on src/main/resources/db/changelog/db.changelog-master.yaml:
```diff
@@ -1,3 +1,17 @@
+  - include:
+      file: db/changelog/changes/07-insert-admin-to-users-table.yaml
+  - include:
+      file: db/changelog/changes/08-insert-admin-to-users_roles-table.yaml
```
```suggestion
      file: db/changelog/changes/08-insert-admin-to-users_roles-table.yaml
      
```

knyazkyylubomir committed:
added and implemented CategoryController;
added CategoryRepository;
added and implemented CategoryServiceImpl;
added swagger to all endpoints;
created Category model;
added Set of categories in Book model;
added method findAllByCategoryId to BookServiceImpl;
added JWT authorization to SecurityConfig

knyazkyylubomir committed:
added minor fix

knyazkyylubomir committed:
Merge pull request #6 from knyazkyylubomir/pr-6

added and implemented CategoryController;
